### ISSUE TITLE
Featured Image label missing after image has been added to page

### ISSUE THREAD

**Issue by andreawetzel** [WordPress/gutenberg#63321]
### Description

Testing 6.6-rc3. After adding a featured image, there is no longer a label above the image stating that it is a Featured Image.

The concern is that someone with little WordPress Experience may inherit a site that has featured images in place and will have no context for what this image is for and how it is related to the page or post. It is also a potential accessibility barrier to not have visual context related to this form input. 

### Step-by-step reproduction instructions

1. Open or create a page
2. Add or upload a featured image

### Screenshots, screen recording, code snippet

![featured-image-no-label](https://github.com/WordPress/gutenberg/assets/9554136/98ec09c7-a8ee-4057-a7f2-754c737db5a1)

### Environment info

WP 6.6-rc3
Twenty TwentyFour theme
Gutenberg plugin not enabled

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Mamaduka** [comment]
cc @jameskoster, @ntsekouras

**Comment by afercia** [comment]
I support adding headings to identify sections of content. The 'label' shouldn't just be a visual label i.e. unsemantic text. It should be a heading element.

The effort to reunify the Post Editor with the Site Editor and the new Post Summary panel introduced many changes and an overall drop in the amount of headings used on the page.

In the screenshot below, a comparison of the headings structure in WordPress 6.5 (left) and the one on trunk (right).
Overall, I wouldn't say the content structure and headings hierarchy has improved, to be fair.

![post panel and headings](https://github.com/WordPress/gutenberg/assets/1682452/36851187-9036-42af-997f-c66da5297044)

I'd like to remind everyone that headings are of fundamental importance for many users to navigate the page and find content.

**Comment by jameskoster** [comment]
I assume we'd also need headings for the other sections (Summary and Excerpt)?

<img width="278" alt="Screenshot 2024-07-10 at 15 50 08" src="https://github.com/WordPress/gutenberg/assets/846565/64ee5df8-159d-4544-9f3b-9743d5249ce3">

cc @WordPress/gutenberg-design for input.

**Comment by afercia** [comment]
> I assume we'd also need headings for the other sections (Summary and Excerpt)?

Ideally, yes but not sure that it should be discussed in the scope of this issue, which is specific to the featured image.

To me, the new Post Summary mixes together settings that are conceptually very different. I'd argue that also from a visual perspecitve, this panel is now a litlte too 'crowded'. Scanning visually its content is now more challenging. This is a broader issue though and there are other things to address e.g.:
- Two H1 headings in the same page
- A H1 and a H2 with the same text (the post title)
- The post title is displayed three times on the page, which is maybe a little too many times?
  1. In the document bar
  2. In the post title field.
  3. In the Post Summary card
- The post settings panel doesn't have a heading that tells uses what it is about e.g. 'Post settings'.
- The verse icon should not be used for hte post title, see https://github.com/WordPress/gutenberg/issues/62186

**Comment by youknowriad** [comment]
Personally, I felt that the new design improved the readability of the sidebar as it became way easier to scan. The old one had way too much information at the same time for my liking and it was also very disorganized. The new one seems more consistent.

There might be some things we can do to improve clarity though but I do appreciate the consistency and the clarity of the new design.

Just my 2cts

**Comment by afercia** [comment]
Besides personal perception and opinions, an UI that aims to be as accessible as possible must provide information. Removing headings or other informative text isn't a great way to improve information. It justs makes the UI _visually lighter_.

An accessible UI should prioritize _function_ over _form_. Which is, I guess, the point where views about design differ between designers and accessibility specialists.

**Comment by youknowriad** [comment]
> An accessible UI should prioritize function over form. 

This is exactly where I'm coming from when I shared the above. It's not about form, it's about the UI being more usable IMO.
